src/OrchardCore.Modules/OrchardCore.Security/Views/SecurityHeadersSettings.Edit.cshtml

```diff
@@ -0,0 +1,359 @@
+@using Microsoft.AspNetCore.Mvc.Localization
+@using OrchardCore.Security.Options
+@using OrchardCore.Security.ViewModels
+@model SecuritySettingsViewModel
+
+@functions
+{
+    public LocalizedHtmlString GetContentSecurityPolicyName(string value)
+    {
+        LocalizedHtmlString name = null;
+        if(value == ContentSecurityPolicyValue.BaseUri)
+        {
+            name = T["Base URI"];
+        }
+        else if(value == ContentSecurityPolicyValue.ChildSource)
+        {
+            name = T["Child Source"];
+        }
+        else if(value == ContentSecurityPolicyValue.ConnectSource)
+        {
+            name = T["Connect Source"];
+        }
+        else if(value == ContentSecurityPolicyValue.DefaultSource)
+        {
+            name = T["Default Source"];
+        }
+        else if(value == ContentSecurityPolicyValue.FontSource)
+        {
+            name = T["Font Source"];
+        }
+        else if(value == ContentSecurityPolicyValue.FormAction)
+        {
+            name = T["Form Action"];
+        }
+        else if(value == ContentSecurityPolicyValue.FrameAncestors)
+        {
+            name = T["Frame Ancestors"];
+        }
+        else if(value == ContentSecurityPolicyValue.FrameSource)
+        {
+            name = T["Frame Source"];
+        }
+        else if(value == ContentSecurityPolicyValue.ImageSource)
+        {
+            name = T["Image Source"];
+        }
+        else if(value == ContentSecurityPolicyValue.ManifestSource)
+        {
+            name = T["Manifest Source"];
+        }
+        else if(value == ContentSecurityPolicyValue.MediaSource)
+        {
+            name = T["Media Source"];
+        }
+        else if(value == ContentSecurityPolicyValue.ObjectSource)
+        {
+            name = T["Object Source"];
+        }
+        else if(value == ContentSecurityPolicyValue.Sandbox)
+        {
+            name = T["Sandbox"];
+        }
+        else if(value == ContentSecurityPolicyValue.ScriptSource)
+        {
+            name = T["Script Source"];
+        }
+        else if(value == ContentSecurityPolicyValue.StyleSource)
+        {
+            name = T["Style Source"];
+        }
+
+        return name;
+    }
+
+    public LocalizedHtmlString GetPermissionPolicyName(string value)
+    {
+        LocalizedHtmlString name = null;
+        if(value == PermissionsPolicyValue.Accelerometer)
```

Comment:
No switch statement?

Reply:
I tried to use pattern matching, but it fails because the values they are not constants, I will do my final touch then I will open it for review